src/java/com/unifina/signalpath/remote/Http.java

```diff
@@ -139,15 +146,27 @@ protected void sendOutput(HttpTransaction call) {
 			try {
 				// entity is null for bodyless response 204, and that's not an error
 				HttpEntity entity = call.response.getEntity();
+
 				if (entity != null) {
-					if (isOctetStream(entity)) {
+					String dt = dataType.getValue();
+
+					if (dt == null) {
+						dt = getDataType(entity);
+					}
+
+					if (dt.equals("binary")) {
```

Comment:
Here's a nifty Java trick for you: When comparing something with Java's equals(), reverse equals comparison in order to eliminate NullPointerException. Like:
`if ("binary".equals(dt)) { // This works even if dt is null`
Code `dt.equals("")` will throw NullPointerException if dt is null!

It's not necessary here because getDataType(String) will return `!`null.